src/include/connection.h

```diff
@@ -287,6 +287,21 @@ typedef TAILQ_HEAD(__wt_backuphash, __wt_backup_target) WT_BACKUPHASH;
 
 extern const WT_NAME_FLAG __wt_stress_types[];
 
+#define WT_SESSION_FOREACH_BEGIN(session, conn)                                                \
+    do {                                                                                       \
+        uint32_t __i, __session_cnt;                                                           \
+        u_int __active;                                                                        \
+        __session_cnt = *(volatile uint32_t *)&(conn)->session_cnt;                            \
```

Comment:
This won't work for hazard pointers, as it is a growable array implementation of the slot counter array pattern. This means we either maintain two versions of this macro or keep this as an ordered read to service both.